### PR TITLE
New Robot HAL

### PR DESCRIPTION
**Pull request by akhilveeraghanta**
### Description
This PR doesn't change _any_ (maybe just a little) of our written code. 

This PR merges functionality with the .ioc file that elec provided, ran code-gen, fixed the build errors and made sure that the board flashed and stuff that was working before still worked.

This PR will break ethernet compatibility with the dev board, but will be restored when elec updates the routing to match the dev board

NOTE: you ONLY  need to review these changes: 

* src/firmware/boards/robot_stm32h7/BUILD
* formatting_scripts/fix_formatting.sh
* src/firmware/boards/robot_stm32h7/freertos.c
* src/firmware/boards/robot_stm32h7/io/BUILD

### Testing Done

Ran the firmware locally on the robot, things run. 

### Resolved Issues
N/A

### Length Justification
N/A Autogenerated code

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Review by amrm3lm — DISMISSED** [review]
lgtm

File: src/firmware/boards/robot_stm32h7/BUILD [hunk 1]
```diff
@@ -8,6 +8,10 @@ cc_binary(
         "freertos.c",
         "main.c",
     ],
+    copts = [
+        # We don't want to mark warnings as errors for 3rd party code
+        "-Wno-error",
+    ],
```
Comment (jonathanlew) [inline src/firmware/boards/robot_stm32h7/BUILD]: Why do we need this now but not in the past?

**Comment by jonathanlew** [comment]
Great to see this coming together!

File: src/firmware/rules.bzl [hunk 2]
```diff
@@ -20,2005 +20,1825 @@ def stm32h7_hal_library_files_genrule(name):
...
-            "Middlewares/Third_Party/LwIP/system/arch/cc.h",
-            "Middlewares/Third_Party/LwIP/system/arch/bpstruct.h",
             "Middlewares/Third_Party/LwIP/system/arch/cpu.h",
+            "Middlewares/Third_Party/LwIP/system/arch/bpstruct.h",
             "Middlewares/Third_Party/LwIP/system/arch/epstruct.h",
-            "Middlewares/Third_Party/LwIP/doc/doxygen/main_page.h",
+            "Middlewares/Third_Party/LwIP/system/arch/init.h",
+            "Middlewares/Third_Party/LwIP/system/arch/perf.h",
+            "Drivers/CMSIS/Device/ST/STM32H7xx/Include/stm32h7xx.h",
+
```
[the remark this answers is not in the document]
Comment (akhilveeraghanta) [reply src/firmware/rules.bzl:1836]: yup its the recommended version for this revision of cube

File: src/firmware/boards/robot_stm32h7/BUILD (hunk 1 above)
Comment (akhilveeraghanta) [reply src/firmware/boards/robot_stm32h7/BUILD]: nuked! used https://gcc.gnu.org/onlinedocs/gcc-4.8.5/gcc/Diagnostic-Pragmas.html around the hal header

File: src/firmware/boards/robot_stm32h7/main.c [hunk 3]
```diff
@@ -36,6 +38,7 @@
 #include "firmware/app/logger/logger.h"
 #include "firmware/boards/robot_stm32h7/io/drivetrain.h"
 #include "firmware/boards/robot_stm32h7/io/uart_logger.h"
+#pragma GCC diagnostic pop
```
Comment (jonathanlew) [inline src/firmware/boards/robot_stm32h7/main.c]: nit:

```suggestion
#pragma GCC diagnostic pop

#include "firmware/app/logger/logger.h"
#include "firmware/boards/robot_stm32h7/io/drivetrain.h"
#include "firmware/boards/robot_stm32h7/io/uart_logger.h"
```

File: src/firmware/boards/robot_stm32h7/io/INA226_power_monitor_driver.h [hunk 4]
```diff
@@ -21,8 +21,13 @@
  * Adapted from: https://github.com/macgeorge/STM32-example-codes
  */
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wsign-conversion"
+#pragma GCC diagnostic ignored "-Wconversion"
```
Comment (jonathanlew) [inline src/firmware/boards/robot_stm32h7/io/INA226_power_monitor_driver.h:26]: suggestion, perhaps for another pr: make a MACRO that disables all warnings, e.g. `-Wall,-Wextra,-Wconversion,-Wno-unused-parameter`

**Review by jonathanlew — COMMENTED** [review]
Looks good just a couple comments

**Review by jonathanlew — APPROVED** [review]
LGTM! Nice work!

File: src/firmware/boards/robot_stm32h7/io/INA226_power_monitor_driver.h (hunk 4 above)
Comment (akhilveeraghanta) [reply src/firmware/boards/robot_stm32h7/io/INA226_power_monitor_driver.h:26]: I'll make a ticket for this, I can't seem to figure out a clean way to do this, looks like nested preprocessor directives aren't easy (maybe I'm just missing something)

Comment (jonathanlew) [reply src/firmware/boards/robot_stm32h7/io/INA226_power_monitor_driver.h:26]: yeah the most obvious way is to have PREPROCESSOR_IGNORE_WARNINGS_START and PREPROCESSOR_IGNORE_WARNINGS_END macros, which isn't particularly elegant

**Review by amrm3lm — APPROVED** [review]

**Pull request merged**